### ISSUE TITLE
Duplicate port allocated

### ISSUE THREAD

**Issue by ndelage** [docker/docker#8847]
Does the random port allocation take IP address of the container into consideration? I ask because I'm running into a situation where one container is bound to `0.0.0.0` and another is bound to the docker0 IP (`172.17.42.1`), but they're both assigned the same port.

This causes issues because `0.0.0.0` acts like a catch-all. If I don't bind the second container to a particular IP and stick with the default of `0.0.0.0`, I don't have this problem (the assigned ports are different).

```
Docker version 1.2.0, build fa7b24f
Execution Driver: native-0.2
Kernel Version: 3.15.4-x86_64-linode45
Operating System: Ubuntu 14.04.1 LTS
```

**Comment by jfrazelle** [comment]
this was fixed in 1.4.0

**Comment by ndelage** [comment]
Wonderful! Thanks for the update.
